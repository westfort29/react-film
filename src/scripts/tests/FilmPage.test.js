import React from 'react';
import { FilmPage } from '../components/FilmPage';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let component;

beforeEach(() => {
  component = shallow(<FilmPage
    sameGenreFilms={[]}
    isFetchingSameGenreFilms={false}
    isErrorFetchinSameGenreFilms={false}
    film={{ genres: [''], release_date: '1.1.2001' }}
    isFetchingFilm={false}
    isErrorFetchingFilm={false}
    findSameGenreFilms={()=>{}}
    getFilmAndSameGenreFilms={()=>{}}/>).instance();
});

global.fetch = jest.fn().mockImplementation( () => new Promise((resolve, reject) => resolve()).catch(e => e));

describe('FilmPage', () => {
  test('should contain right data', () => {
    const renderedComponent = renderer.create(<FilmPage
      sameGenreFilms={[]}
      isFetchingSameGenreFilms={false}
      isErrorFetchinSameGenreFilms={false}
      film={{ genres: [''], release_date: '1.1.2001' }}
      isFetchingFilm={false}
      isErrorFetchingFilm={false}
      findSameGenreFilms={()=>{}}
      getFilmAndSameGenreFilms={()=>{}}/>);
    let tree = renderedComponent.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
