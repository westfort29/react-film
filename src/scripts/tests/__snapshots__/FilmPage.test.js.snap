// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPage should contain right data 1`] = `
<div
  className="film-page"
>
  <div
    className="film-page__info-wrapper"
  >
    <div
      className="film-page__content"
    >
      <div
        className="film-page__heading"
      >
        <span
          className="logo"
        >
          netflixroulette
        </span>
        <a
          className="film-page__search-link"
        >
          Search
        </a>
      </div>
      <section
        className="film-page__film-info"
      >
        <div
          className="film-page__poster-wrapper"
        >
          <img
            className="film-page__poster"
          />
        </div>
        <div
          className="film-page__text-info"
        >
          <h1
            className="film-page__film-title"
          />
          <p
            className="film-page__film-rewards"
          />
          <p
            className="film-page__film-time"
          >
            <span
              className="film-page__film-year"
            >
              2001
            </span>
            <span
              className="film-page__film-duration"
            >
               min
            </span>
          </p>
          <p
            className="film-page__film-description"
          />
        </div>
      </section>
    </div>
  </div>
  <div
    className="film-page__same-genre-heading"
  >
    Films by 
    <span
      className="film-page__genre-title"
    >
      
       genre 
    </span>
  </div>
  <ul
    className="films-search-result"
  >
    <li
      className="films-search-result__film films-search-result__film--dummy"
    />
    <li
      className="films-search-result__film films-search-result__film--dummy"
    />
    <li
      className="films-search-result__film films-search-result__film--dummy"
    />
    <li
      className="films-search-result__film films-search-result__film--dummy"
    />
  </ul>
</div>
`;
